aptos-move/framework/aptos-framework/sources/jwks.move

```diff
@@ -0,0 +1,480 @@
+    const EUNKNOWN_JWK_MAP_EDIT: u64 = 3;
+
+    /// An OIDC provider.
+    struct OIDCProvider has drop, store {
+        name: vector<u8>, // e.g., b"https://www.facebook.com"
+        config_url: vector<u8>, // e.g., b"https://www.facebook.com/.well-known/openid-configuration/"
+    }
+
+    /// The OIDC provider set. Maintained by governance proposals.
+    struct OIDCProviderSet has key {
+        providers: vector<OIDCProvider>, // The order does not matter.
+    }
+
+    /// Some extra configs that controls JWK consensus behavior. Maintained by governance proposals.
+    ///
+    /// Currently supported `content` types:
+    /// - `JWKConsensusConfigV0`
+    struct JWKConsensusConfig has drop, key {
+        content: Any,
+    }
+
+
+    struct JWKConsensusConfigV0 has copy, drop, store {
+        observation_interval_ms: u64,
+    }
+
+    /// An observed but not yet supported JWK.
+    struct UnsupportedJWK has copy, drop, store {
+        payload: vector<u8>,
+    }
+
+    /// A JWK where `kty` is `RSA`.
+    struct RSA_JWK has copy, drop, store {
+        kid: String,
+        kty: String,
+        alg: String,
+        e: String,
+        n: String,
+    }
+
+    /// A JWK.
+    ///
+    /// Currently supported `content` types:
+    /// - `RSA_JWK`
+    /// - `UnsupportedJWK`
+    struct JWK has copy, drop, store {
+        id: vector<u8>,
+        content: Any,
+    }
+
+    /// A provider and its JWKs.
+    struct ProviderJWKSet has copy, drop, store {
+        /// The utf-8 encoding of the issuer string (e.g., "https://accounts.google.com").
+        issuer: vector<u8>,
+
+        /// The `jwks` should each have a unique `id`, and should be sorted by `id` in alphabetical order.
+        jwks: vector<JWK>,
+    }
+
+    /// All OIDC providers and their JWK sets.
+    struct JWKMap has copy, drop, store {
+        /// Entries should each have a unique `issuer`, and should be sorted by `issuer` in an alphabetical order.
+        entries: vector<ProviderJWKSet>,
+    }
+
+    /// A `JWKMap` maintained by JWK consensus.
+    struct OnChainJWKMap has copy, drop, key, store {
+        version: u64,
+        jwk_map: JWKMap,
+    }
+
+    #[event]
+    /// When an on-chain JWK set update is done, this event is sent to reset the JWK consensus state in all validators.
+    struct OnChainJWKMapUpdated has drop, store {
+        epoch: u64,
+        version: u64,
+        jwk_map: JWKMap,
+    }
+
+    /// A small edit that can be applied to a `JWKMap`.
+    ///
+    /// Currently supported `content` types:
+    /// - `JWKMapEditCmdDelAll`
+    /// - `JWKMapEditCmdDelIssuer`
+    /// - `JWKMapEditCmdDelJwk`
+    /// - `JWKMapEditCmdPutJwk`
+    struct JWKMapEdit has drop, store {
+        content: Any,
+    }
+
+    struct JWKMapEditCmdDelAll has copy, drop, store {}
+
+    struct JWKMapEditCmdDelIssuer has copy, drop, store {
+        issuer: vector<u8>,
+    }
+
+    struct JWKMapEditCmdDelJwk has copy, drop, store {
```

Comment:
I would rename all of these commands struct to be easier to read...

For example, just call this `DeleteJwkOp`. Remove the whole `JWKMapEditCmd` prefix... it's way too long.

Do the same for the others.

Reply:
Regarding the prefix, it helps these `cmd` structs to be listed close to each other in `jwks.md` so when people look at it they know they are related.
Naming like `DeleteJwkOp` will result in random positions which is uncool...

Also all specific `cmd` structs are internal only. I will add comments.

Reply:
Then, do `OpDeleteJwk`